src/jbms/argparse.cpp

```diff
@@ -19,15 +19,116 @@ namespace argparse {
+  } else if ((ch >= 32 && ch <= 126) && ch != '\'' && ch != '\"' &&
+             ch != '\\') {
+    sprintf(
+        buffer, "%c",
+        ch);  // opposite of control characters can be saved just as they are
+  } else if (buflen < 3) {
+    *buffer = '\0';  // just add a null-terminator; don't need to go further
+  } else {
+    switch (ch) {
+      case '\a':
+        strcpy(buffer, "\\a");
+        break;
+      case '\b':
+        strcpy(buffer, "\\b");
+        break;
+      case '\f':
+        strcpy(buffer, "\\f");
+        break;
+      case '\n':
+        strcpy(buffer, "\\n");
+        break;
+      case '\r':
+        strcpy(buffer, "\\r");
+        break;
+      case '\t':
+        strcpy(buffer, "\\t");
+        break;
+      case '\v':
+        strcpy(buffer, "\\v");
+        break;
+      case '\\':
+        strcpy(buffer, "\\\\");
+        break;
+      case '\'':
+        strcpy(buffer, "\\'");
+        break;
+      case '\"':
+        strcpy(buffer, "\\\"");
+        break;
+      default:
+        if (buflen < 5)
+          *buffer = '\0';
+        else
+          sprintf(buffer, "\\%03o",
+                  ch);  // this handles control characters that are not
+                        // printable and does not meet any case
+        break;
+    }
+  }
 }
 
+void str_cstrliteral(const std::string &str, std::string &buffer,
+                     std::size_t reserved_bits) {
+  /**
+   * @brief Escapes the special string @p str and stores inside of the @p
+   * buffer.
+   * @param str Special characters string.
+   * @param buffer Buffer to store the escaped value in.
+   * @param buflen Length of the buffer.
+   */
+  // FIXME: Make this escape NULL bytes as well
+  for (const auto &ch : str) {
+    char *buf =
+        new char[buffer.length() +
+                 1];  // need to create a buffer inorder to write to the
+                      // memory inside the above function i.e., chr_cstrliteral()
+    strcpy(buf, buffer.c_str());  // copy the content of the buffer inside the
+                                  // memory that we just reserved
+    chr_cstrliteral(ch, buf + buffer.length(),
+                    reserved_bits);  // escape the character that we took out in
+                                     // this iteration
+    reserved_bits -=
+        buffer.length();  // impede the 'reserved_bits' by the buffer length
+    buffer = buf;         // copy inside the actual buffer
+    delete[] buf;         // free the free-store
+  }
+}
+}  // namespace
+
+std::string repr(std::string str) {
+  /**
+   * @brief Escape ASCII control characters.
+   * @param str String to escape.
+   * @returns Escaped string.
+   */
+  std::size_t reserved_bits =
+      str.length() * 2;  // maximum length that the string
+                         // containing special characters can have
+  std::string buffer;
+  buffer.reserve(reserved_bits);  // reserve 'reserved_bits' bytes for buffer
```

Comment:
Still need to reserve space in-order to avoid the following exception that occurs inside the function `str_cstrliteral()`.
```shell
free(): invalid pointer
Aborted (core dumped)
```